staging/src/k8s.io/metrics/pkg/client/custom_metrics/client.go

```diff
@@ -20,27 +20,52 @@ import (
 	"fmt"
 
 	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	serializer "k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/discovery"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/util/flowcontrol"
+
+	cmint "k8s.io/metrics/pkg/apis/custom_metrics"
+	"k8s.io/metrics/pkg/apis/custom_metrics/v1beta1"
 	"k8s.io/metrics/pkg/apis/custom_metrics/v1beta2"
 	"k8s.io/metrics/pkg/client/custom_metrics/scheme"
 )
 
+var metricConverter = NewMetricConverter()
+var codecs = serializer.NewCodecFactory(metricConverter.Scheme())
+
+type AvailableMetricsAPIFunc func() (*metav1.APIGroupList, error)
+
+type APIVersionsFromDiscovery struct {
+	discovery.DiscoveryClient
+}
+
+func (d APIVersionsFromDiscovery) Versions() (*metav1.APIGroupList, error) {
+	groups, err := d.DiscoveryClient.ServerGroups()
```

Comment:
this actual implementation should be filtering the groups by the metrics API -- it shouldn't be down the code in the metrics client.  This allows us to test the two independently.

Reply:
(i.e. I'd lift the "find the metrics API group" out of the converter).